apps/web/src/utils/skillUtils.ts

```diff
@@ -330,11 +330,31 @@ export const sortPoolSkillsBySkillCategory = <T extends PoolSkill[]>(
  * @returns Skill[]
  */
 export const filterPoolSkillsByType = (
-  poolSkills: PoolSkill[],
+  poolSkills: Maybe<Maybe<PoolSkill>[]> | undefined,
```

Comment:
This is such an ugly type 🤣

Reply:
I've seen worse, had to implement this as a workaround. Will be gone with https://github.com/GCTC-NTGC/gc-digital-talent/issues/9901

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/04d4b3d6-0df7-4ff3-9edd-e2bc4347f12f)

Reply:
Yeah :sweat_smile: The more I work on these queries and masking the more I want to start re-writing everything to use the masksing. I really like how it basically writes your types for you :sunglasses: